File: backend/UniUti/models/Disciplina.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using UniUti.Models.Base;

namespace UniUti.Models
{
    [Table("disciplinas")]
    public class Disciplina : EntidadeBase
    {
        [Column("nome")]
        [Required]
        [StringLength(100)]
        public string? Nome { get; set; }

        [Column("descricao")]
        [Required]
        [StringLength(300)]
        public string? Descricao { get; set; }

        public virtual ICollection<Curso>? Cursos { get; set; }

        [Column("deletado")]
        public Boolean Deletado { get; set; } = false;
    }
}

File: uniuti_core/lib/disciplina/domain/disciplina.dart
part of '../../uniuti_core.dart';

class Disciplina {
  int id;
  String nome;
  String descricao;
  String periodicidade;
  Disciplina({
    required this.id,
    required this.nome,
    required this.descricao,
    this.periodicidade = '',
  });
}
